src/org/opensolaris/opengrok/history/HistoryCache.java

```diff
@@ -126,6 +126,12 @@ String getLatestCachedRevision(Repository repository)
     void clear(Repository repository) throws HistoryException;
 
     /**
+     * Clear entry for single file from history cache.
+     * @param file path to the file
+     */
+    void clearFile(String file);
```

Comment:
the path is relative to source root. It cannot be a `File` object because that would bypass HistoryGuru as a generic interface to concrete history cache backend (remember the days when OpenGrok supported JavaDB history cache next to File history cache).